src/ray/raylet/node_manager.cc

```diff
@@ -957,6 +1056,33 @@ void NodeManager::HandleObjectMissing(const ObjectID &object_id) {
   }
 }
 
+void NodeManager::ForwardTaskOrResubmit(const Task &task,
+                                        const ClientID &node_manager_id) {
+  /// TODO(rkn): Should we check that the node manager is remote and not local?
+  /// TODO(rkn): Should we check if the remote node manager is known to be dead?
+  const TaskID task_id = task.GetTaskSpecification().TaskId();
+
+  // Attempt to forward the task.
+  if (!ForwardTask(task, node_manager_id).ok()) {
+    RAY_LOG(INFO) << "Failed to forward task " << task_id << " to node manager "
+                  << node_manager_id;
+
+    // Create a timer to resubmit the task in a little bit.
+    boost::asio::deadline_timer timer(io_service_);
+    auto retry_duration = boost::posix_time::milliseconds(
+        RayConfig::instance().node_manager_forward_task_retry_timeout_milliseconds());
+    timer.expires_from_now(retry_duration);
+    timer.async_wait([this, task, task_id](const boost::system::error_code &error) {
```

Comment:
Hmm have you tried out this codepath? I think this actually won't work since the `boost::asio::deadline_timer` will expire immediately if it goes out of scope. You'll probably want to dynamically allocate a timer with `unique_ptr` and then store them as a set, or something along those lines.

Reply:
Oh thanks for pointing that out, I wasn't sure about the timer behavior. I did verify that this prevented the node manager from dying when attempting to forward a task to a dead node manager, but that's all.

By the way, I essentially want the timer to deallocate itself when it fires.. is there a clean way to do that (this seems like it must be a common use case..)?

Reply:
Unfortunately, not that I know of...maybe eventually it would make sense for us to pool the timers, but for now we'll probably just have to do allocation/deallocation on the fly.